Source/missiles.cpp

```diff
@@ -915,6 +874,105 @@ Direction16 GetDirection16(Point p1, Point p2)
 	return ret;
 }
 
+void MissileHitMonsterConsequences(int mid, int pnum, int dam, missile_id mName)
+{
+	auto &monster = Monsters[mid];
+	const auto &player = Players[pnum];
+
+	bool hasKnockback = false;
+	if (pnum >= 0) {
+		const auto &player = Players[pnum];
+		if (HasAnyOf(player._pIFlags, ItemSpecialEffect::Knockback))
+			hasKnockback = true;
+	}
+
+	if (monster._mhitpoints >> 6 <= 0) {
+		M_StartKill(mid, pnum);
+	} else if (monster.IsResistant(mName)) {
+		PlayEffect(monster, 1);
+	} else {
+		if (monster._mmode != MonsterMode::Petrified
+		    && MissilesData[mName].mType == 0
+		    && hasKnockback)
+			M_GetKnockback(mid);
+		if (monster.MType->mtype != MT_GOLEM)
+			M_StartHit(mid, pnum, dam);
+	}
+}
+
+int PlayerMissileCalculateCTHAgainstMonster(int pnum, missile_id mName, int dist, Monster &monster)
+{
+	int hper = 0;
+	if (pnum != -1) {
+		const auto &player = Players[pnum];
+		if (MissilesData[mName].mType == 0) {

```

Comment:
Not necessary for this PR, but adding it since it took me a bit of time to figure out. This check is to see if it's a physical arrow or a magic spell. It would be best replaced with something like:
```cpp
bool Missile::isArrow()
{
	IsAnyOf(_mitype, MIS_ARROW, MIS_FARROW, MIS_LARROW, MIS_SPECARROW);
}
```
(`MIS_KRULL` is technically also a physical arrow, but is unused and can be dropped)

With that `mType` could be doped from the `MissileData` struct as well.

Reply:
You can freely change `MissileData` as it is static data that does not affect the save games.

Reply:
> You can freely change `MissileData` as it is static data that does not affect the save games.

right, I thought somehow about Missile struct :) but its not there.